torch/csrc/jit/codegen/cuda/ir_internal_nodes.h

```diff
@@ -549,6 +549,36 @@ class TORCH_CUDA_CU_API ViewOp : public Expr {
   TensorView* const in_ = nullptr;
 };
 
+//! This operator explicitly models data movement between
+//!   state spaces on GPU. Currently the modeled state spaces include
+//!   global memory, shared memory and register.
+//!
+//! The main usage of this op is to facilitate generation of hardware
+//!   accelerated memory ops, i.e. ldmatrix, cp.async and more to come.
+class TORCH_CUDA_CU_API LoadStoreOp : public Expr {
```

Comment:
I wonder if it would make sense if this inherited from `UnaryOp`? It is a `UnaryOp`, so conceptually it should make sense, although the expression classes are not really structured with a semantic class hierarchy, e.g., `BroadcastOp` is also a `UnaryOp` but does not inherit from `UnaryOp`. 

Note that the new reduction class, `GroupedReductionOp`, inherits from `ReductionOp` as I find it more convenient.

Reply:
Sure, would be interesting to explore in follow ups. Currently it didn't seem so clear to me between `LoadStoreOp` and `UnaryOp`, and inheritance didn't seem to benefit dispatching that much with the current implementation. 

Additionally, I was still hoping that one day we could make `isA` not depend on dynamic_cast, as it is also used on hot paths, and these are kind of all related in the way we setup the IR nodes.

Reply:
I don't think it matters much for dispatching. It'd be convenient if we wanted to do something for any unary expression as we could write something like:

```
if (expr->isA<UnaryOp>()) {
   ...
}
```

And this would work for both the set op and `LoadStoreOp`. 

That said, I don't think inheritance is used more intensively as the current design of IR nodes is mostly working just fine.